cirq-core/cirq/linalg/transformations.py

```diff
@@ -717,6 +717,11 @@ def factor_density_matrix(
         t2 = transpose_density_matrix_to_axis_order(t1, product_axes)
         if not np.allclose(t2, t, atol=atol):
             raise ValueError('The tensor cannot be factored by the requested axes')
+    # Tweak trace back to 1 to avoid rounding error propagation
+    extracted_dim = int(np.sqrt(np.prod(extracted.shape)))
+    extracted /= np.trace(extracted.reshape((extracted_dim, extracted_dim)))
```

Comment:
Open question: should this go into `partial_trace` instead? Since it appears that `partial_trace` is the root cause of the rounding error.